packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -493,14 +472,51 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		localId: string,
 		blob: ArrayBufferLike,
 	): Promise<ICreateBlobResponse | void> {
-		return PerformanceEvent.timedExecAsync(
+		return runWithRetry(
+			async () => {
+				try {
+					return await this.getStorage().createBlob(blob);
+				} catch (error) {
+					const entry = this.pendingBlobs.get(localId);
+					assert(
+						!!entry,
+						0x387 /* Must have pending blob entry for blob which failed to upload */,
+					);
+					if (entry.opsent && !canRetryOnError(error)) {
+						if (isFluidError(error)) {
+							error.addTelemetryProperties({ forceRetry: true });
+							throw error;
+						} else {
+							throw wrapError(error, () =>
+								createGenericNetworkError(
+									`uploadBlob error`,
+									{ canRetry: true },
+									{
+										driverVersion: undefined,
+									},
+								),
+							);
+						}
+					}
+					if (canRetryOnError(error)) {
```

Comment:
this should just throw, and the delete should happen in a .catch outside of the retry